src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.CompilationTracker.cs

```diff
@@ -433,11 +445,11 @@ public Task<Compilation> GetCompilationAsync(SolutionCompilationState compilatio
             private async Task<Compilation> GetCompilationSlowAsync(
                 SolutionCompilationState compilationState, CancellationToken cancellationToken)
             {
-                var compilationInfo = await GetOrBuildCompilationInfoAsync(compilationState, lockGate: true, cancellationToken: cancellationToken).ConfigureAwait(false);
-                return compilationInfo.Compilation;
+                var finalState = await GetOrBuildFinalStateAsync(compilationState, lockGate: true, cancellationToken: cancellationToken).ConfigureAwait(false);
+                return finalState.FinalCompilationWithGeneratedDocuments;
             }
 
-            private async Task<CompilationInfo> GetOrBuildCompilationInfoAsync(
```

Comment:
the CompilationInfo type seemed to serve no purpose.  It just grabbed 3 pieces of state *only* out of FinalState, only to then have the callers grab the one piece of data they needed.  So now it's gone and we just use the FinalState value itself.